src/core/crypto/ecdsa.cpp

```diff
@@ -86,8 +96,13 @@ Error P256::KeyPair::Parse(void *aContext) const
     mbedtls_pk_init(pk);
 
     VerifyOrExit(mbedtls_pk_setup(pk, mbedtls_pk_info_from_type(MBEDTLS_PK_ECKEY)) == 0, error = kErrorFailed);
+#if (MBEDTLS_VERSION_NUMBER >= 0x03000000)
+    VerifyOrExit(mbedtls_pk_parse_key(pk, mDerBytes, mDerLength, nullptr, 0, mbedtls_ctr_drbg_random,
+                                      Random::Crypto::MbedTlsContextGet()) == 0,
+                 error = kErrorParse);
+#else
     VerifyOrExit(mbedtls_pk_parse_key(pk, mDerBytes, mDerLength, nullptr, 0) == 0, error = kErrorParse);
-
+#endif
 exit:
     return error;
 }
```

Comment:
About accessing the coordinates of a point directly as a bignum in `P256::KeyPair::GetPublicKey`: I'm not sure that we want to allow that, for the same reason discussed in https://github.com/ARMmbed/mbedtls/issues/4838#issuecomment-892918852. This will require a design discussion in the Mbed TLS team.